setup/setup_rirs.py

```
@@ -0,0 +1,406 @@
+        except Exception as e:
+            print(f"Error downloading {url}: {e}")
+            return False
+
+    def extract_archive(self, archive_path: Path, extract_to: Path) -> bool:
+        """Extract zip or tar.gz archive"""
+        try:
+            print(f"Extracting {archive_path} to {extract_to}...")
+
+            if archive_path.suffix.lower() == '.zip':
+                with zipfile.ZipFile(archive_path, 'r') as zip_ref:
+                    zip_ref.extractall(extract_to)
```

Ruff (0.13.1)

Uses of `tarfile.extractall()`

(S202)